### PR TITLE
fix: stage artifact logic 

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

This PR fixes the issue: if any task gets fail in pre/post CI/CD we are considering that we have no artifcats to download

Fixes https://github.com/devtron-labs/devtron/issues/5535

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [x] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,6 +6,8 @@
 RUN go install github.com/google/wire/cmd/wire@latest
 WORKDIR /go/src/github.com/devtron-labs/devtron
 ADD . /go/src/github.com/devtron-labs/devtron/
+# TODO Asutosh: revert after dev test
+ADD ./vendor/github.com/Microsoft/ /go/src/github.com/devtron-labs/devtron/vendor/github.com/microsoft/
```
[the remark this answers is not in the document]
Comment (gireesh-naidu) [reply Dockerfile:10]: please look at this

**Review by gireesh-naidu — DISMISSED** [review]

File: pkg/pipeline/CiService.go [hunk 2]
```diff
@@ -487,26 +480,20 @@ func (impl *CiServiceImpl) executeCiPipeline(workflowRequest *types.WorkflowRequ
...
-	ciArtifactLocationFormat := ciWorkflowConfig.CiArtifactLocationFormat
-	if ciArtifactLocationFormat == "" {
-		ciArtifactLocationFormat = impl.config.GetArtifactLocationFormat()
-	}
-	ArtifactLocation := fmt.Sprintf("s3://%s/%s/"+ciArtifactLocationFormat, ciWorkflowConfig.LogsBucket, impl.config.GetDefaultArtifactKeyPrefix(), savedWf.Id, savedWf.Id)
-	artifactFileName := fmt.Sprintf(impl.config.GetDefaultArtifactKeyPrefix()+"/"+ciArtifactLocationFormat, savedWf.Id, savedWf.Id)
-	return ArtifactLocation, ciWorkflowConfig.LogsBucket, artifactFileName
+func (impl *CiServiceImpl) buildS3ArtifactLocation(ciWorkflowConfigLogsBucket string, savedWf *pipelineConfig.CiWorkflow) (string, string, string) {
+	ciArtifactLocationFormat := impl.config.GetArtifactLocationFormat()
+	ArtifactLocation := fmt.Sprintf("s3://"+path.Join(ciWorkflowConfigLogsBucket, ciArtifactLocationFormat), savedWf.Id, savedWf.Id)
```
Comment (nishant-d) [inline pkg/pipeline/CiService.go:485]: duplicate

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ash-exp) [reply Dockerfile]: updated 👍

File: pkg/pipeline/CiService.go (hunk 2 above)
Comment (Ash-exp) [reply pkg/pipeline/CiService.go:485]: there are two methods `buildS3ArtifactLocation` and `buildArtifactLocationForS3`. But, both of them serve different business logics.

**Review by vikramdevtron — DISMISSED** [review]

**Review by vikramdevtron — APPROVED** [review]

**Pull request merged**